test: mark flaky tests as flaky

joyeecheung opened the pull request:
In the last 100 CI runs, these tests have failed too many unrelated PRs

```
Reason     async-hooks/test-callback-error
Type       JS_TEST_FAILURE
Failed PR  13 (#22266, #22211, #22274, #22289, #22284, #22293, #22300, #22273, #22310, #22249, #22306, #22318, #22269)
Appeared   test-digitalocean-fedora27-x64-1
First CI   https://ci.nodejs.org/job/node-test-pull-request/16383/
Last CI    https://ci.nodejs.org/job/node-test-pull-request/16450/
--------------------------------------------------------------------------------
Reason     parallel/test-cluster-master-error
Type       JS_TEST_FAILURE
Failed PR  10 (#22293, #22307, #22310, #22306, #22249, #22320, #22318, #21611, #22128, #22269)
Appeared   test-joyent-alpine38_container-x64-1, test-softlayer-alpine38_container-x64-1
First CI   https://ci.nodejs.org/job/node-test-pull-request/16434/
Last CI    https://ci.nodejs.org/job/node-test-pull-request/16450/
--------------------------------------------------------------------------------
Reason     parallel/test-cluster-master-kill
Type       JS_TEST_FAILURE
Failed PR  10 (#22293, #22307, #22310, #22306, #22249, #22320, #22318, #21611, #22128, #22269)
Appeared   test-joyent-alpine38_container-x64-1, test-softlayer-alpine38_container-x64-1
First CI   https://ci.nodejs.org/job/node-test-pull-request/16434/
Last CI    https://ci.nodejs.org/job/node-test-pull-request/16450/
--------------------------------------------------------------------------------
Reason     async-hooks/test-statwatcher
Type       JS_TEST_FAILURE
Failed PR  7 (#22274, #22290, #22288, #22293, #22242, #21611, #22302)
Appeared   test-azure_msft-win2016-x64-4, test-joyent-ubuntu1604_sharedlibs_container-x64-1
First CI   https://ci.nodejs.org/job/node-test-pull-request/16387/
Last CI    https://ci.nodejs.org/job/node-test-pull-request/16451/
```

Mark them as flaky for now so we can land PRs

joyeecheung commented:
Waiting on the container rebuilds

Trott commented on test/parallel/parallel.status:
```diff
@@ -9,6 +9,9 @@ prefix parallel
 [$system==linux]
+# https://github.com/nodejs/node/issues/22308
+test-cluster-master-kill: PASS,FLAKY
+test-cluster-master-error: PASS,FLAKY
```
The host/platform that these were failing on has been removed, so I think there's no need to mark them as flaky. (For what it's worth, they were failing 100% of the time and I'm not a fan of marking 100% failures as flaky.)

rvagg replied:
On the other hand, I'd like to see Alpine 3.8 back in rotation. We're risking either ignoring Alpine 3.8 completely, or if we mark this as flaky we risk not fixing the underlying problem. Pick your poison.

refack replied:
I'm also against adding this, since it'll create a permanent yellow CI, and that is desensitizing...
We have a tracking issue, and we have eyes on the problem, so the chances of loosing track are lower.

rvagg commented:
and again https://ci.nodejs.org/job/node-test-pull-request/16457/
and https://ci.nodejs.org/view/All/job/node-test-commit-alpine38/5/ for Alpine 3.8 specifically

joyeecheung commented:
Removed the cluster errors. New CI: https://ci.nodejs.org/job/node-test-pull-request/16461/

joyeecheung commented:
Failures in job https://ci.nodejs.org/job/node-test-pull-request/16461/

#### [ubuntu1604-arm64](https://ci.nodejs.org/job/node-test-commit-arm/nodes=ubuntu1604-arm64/17731/console)

<details>
<summary>See failures on test-packetnet-ubuntu1604-arm64-2:</summary>

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   deps/v8/third_party/jinja2/LICENSE

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	build/

no changes added to commit
```
</details>

#### [ubuntu1604_sharedlibs_debug_x64](https://ci.nodejs.org/job/node-test-commit-linux-containered/nodes=ubuntu1604_sharedlibs_debug_x64/6280/console)

<details>
<summary>See failures on test-softlayer-ubuntu1604_sharedlibs_container-x64-2:</summary>

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   deps/v8/third_party/jinja2/LICENSE

no changes added to commit
```
</details>

#### [ubuntu1604_sharedlibs_openssl110_x64](https://ci.nodejs.org/job/node-test-commit-linux-containered/nodes=ubuntu1604_sharedlibs_openssl110_x64/6280/console)

<details>
<summary>See failures on test-softlayer-ubuntu1604_sharedlibs_container-x64-5:</summary>

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   deps/v8/third_party/jinja2/LICENSE

no changes added to commit
```
</details>


All the failures are git failures, so this should be ready to land

joyeecheung commented:
Landed in b87cb5da44f219dfcd2e2c4757eb5a206ef3b702 and 59c8abf8862dfe0df8d19cee0de9597388e43737

Trott commented:
> All the failures are git failures, so this should be ready to land

Do note that the [COLLABORATOR_GUIDE "Testing and CI" section](https://github.com/nodejs/node/blob/master/COLLABORATOR_GUIDE.md#testing-and-ci) says that those need to be fixed and re-run rather than disregarded and landed:

> Do not land any Pull Requests without passing (green or yellow) CI runs. If you believe any failed (red or grey) CI sub-tasks are unrelated to the change in the Pull Request, use "Resume Build" in the left navigation of the relevant `node-test-pull-request` job. It will create a new node-test-pull-request run that preserves all the green results from the current job but re-runs everything else.

joyeecheung commented:
@Trott Ah, sorry about that, although in this case resuming the builds does not help because those are build failures. I am going to fix them in https://github.com/nodejs/build/issues/1453
